src/hooks/useLocations.js

```diff
@@ -15,30 +16,48 @@ export const useLocations = (lat, lng) => {
     const baseURL = 'https://segdeha.com/api/nearby.php';
     const completeURL = lat && lng && `${baseURL}?lat=${lat}&lng=${lng}`;
 
-    setStatus('loading');
-    fetch(completeURL, { signal })
-      .then((response) => {
-        setIsFetching(true);
-        return response.json();
-      })
-      .then((data) => {
-        setLocations(data?.query?.pages);
-        setStatus('success');
-      })
-      .catch((err) => {
-        if (err.name !== 'AbortError') {
-          setStatus('error');
-          setError(err);
-        }
-      })
-      .finally(() => setIsFetching(false));
-
-    // Clean up
-    return function cancel() {
-      setStatus('cancelled');
-      abortController?.abort();
+    const fetchLocations = (lat, lng) => {
+      setStatus('loading');
+      fetch(completeURL, { signal })
+        .then((response) => {
+          setIsFetching(true);
+          return response.json();
+        })
+        .then((data) => {
+          setLocations(data?.query?.pages);
+          setStatus('success');
+        })
+        .catch((err) => {
+          if (err.name !== 'AbortError') {
+            setStatus('error');
+            setError(err);
+          }
+        })
+        .finally(() => setIsFetching(false));
+
+      //Clean up
+      return function cancel() {
+        setStatus('cancelled');
+        abortController?.abort();
+      };
     };
-  }, [lat, lng]);
+
+    if (isDragged) {
+      fetchLocations(lat, lng);
+      return;
+    }
+
+    if (Date.now() >= ref.current + interval) {
+      ref.current = Date.now();
+      fetchLocations(lat, lng);
+    } else {
+      const id = setTimeout(() => {
+        ref.current = Date.now();
+        fetchLocations(lat, lng);
+      }, interval - (Date.now() - ref.current));
```

Comment:
Curiosity & Doubts:

I'm honest I couldn't understand line 57. I understood that `interval` is your delay time but... 

1. Is it not enough `interval`? Why do you do additional math operations? 
2. Also, I saw that you pass the interval like `interval = wait * 1000` (I guess you want to pass delay in seconds)

Reply:
1) Let’s say our wait period is 10 seconds. This bit of code is here to make sure we only have delays of up to 10 seconds  and that we don’t delay API calls for more than 10 seconds 
2) We want to make sure useLocations gets value in ``milliseconds``, and it’s why we’re multiplying it by 1000